src/initializers/joi.ts

```diff
@@ -190,7 +190,8 @@ const Joi: JoiWithExtensions = _Joi.extend(
           }
         ],
         validate(value, helpers, {bucket}) {
-          if (!isOwnS3Path(bucket, value)) {
+          const { hostname, protocol, pathname} = new URL(value);
+          if (!isOwnS3Path(bucket, `${protocol}//${hostname}${pathname}`)) {
```

Comment:
Clever! 💯  
I think we could also simplify it by doing
```suggestion
          const { origin } = new URL(value);
          if (!isOwnS3Path(bucket, origin)) {
```

Reply:
As talked offline we need the path name also to make the validation because s3 can have the bucket in the path after domain